pallets/dao/src/lib.rs

```diff
@@ -543,9 +502,12 @@ pub mod pallet {
 						org.description = d;
 					}
 					if let Some(v) = vision {
-						org.vision =v;
+						org.vision.vision_literal = v;
+						org.vision.updated_on = current_block;
+						org.vision.updated_by = owner;
 					}
-					org.last_updated = <frame_system::Pallet<T>>::block_number();
+
+					org.last_updated = current_block;
 					Ok(())
 				} else {
 					Err(Error::<T>::InvalidOrganization.into())
```

Comment:
On line 530 below - should we not check to see if organization exists/existed before decreasing the count? E.g. removing the same org_id multiple times would mean the org count is out of sync.

Reply:
i have moved the checks to be before storage writes, nice spot

Reply:
https://github.com/UniversalDot/universal-dot-node/pull/82/commits/70c03868d3dbde4e0ce567485c7032dd6d48eab5